### PR TITLE
WINDUPRULE-635: spring shell to quarkus extension rule

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.redhat.com/browse/WINDUPRULE-635

### CODE REVIEW

File: rules-reviewed/quarkus/springboot/springboot-shell-to-quarkus.windup.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    </metadata>
+    <rules>
+        <rule id="springboot-shell-to-quarkus-00000">
+            <when>
+                <project>
+                    <artifact groupId="org.springframework.shell" artifactId="spring-shell-core"/>
+                </project>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Shell artifact with Quarkus 'picocli' extension" effort="1" category-id="mandatory">
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-shell-to-quarkus.windup.xml:24]: I think it's needed a further investigation. 
Although Quarkus has the command line mode with `picocli` the migration is not direct as it doesn't reuse the Spring Interfaces, so it's needed to re-code.
For instance :  
* Spring Shell = @ShellComponent, @ShellMethod, @ShellOption
* PicoCli = @CommandLine.Command, @CommandLine.Option, @TopCommand

https://quarkus.io/guides/picocli
https://spring.io/projects/spring-shell

**Comment by m-brophy** [comment]
@jonathanvila similar issues requiring re-coding seem to have effort="3" and I've altered the message and put in the spring-shell link

File: rules-reviewed/quarkus/springboot/springboot-shell-to-quarkus.windup.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    <message>
+                        Replace the SpringBoot Shell artifacts with Quarkus `picocli` extension
+
+                        Add Quarkus dependency `io.quarkus:quarkus-picocli`
+
+                        PicoCli provides command-line mode like Spring Shell but does not reuse the Spring interfaces
+                        therefore re-coding is also required to use the appropriate PicoCli interfaces.
+                    </message>
+                    <link title="Quarkus Command Mode with Picocli Guide" href="https://quarkus.io/guides/picocli" />
+                    <link title="Spring Shell Guide" href="https://spring.io/projects/spring-shell" />
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-shell-to-quarkus.windup.xml:34]: Good point !!! Adding documentation to spring :+1:

**Comment by PhilipCattanach** [comment]
The rule works fine, the message is well worded, all of the links work and are appropriate. All good from my perspective.

**Pull request merged**